src/Cosi-BLS/new-transactions.lisp

```diff
@@ -1378,14 +1378,14 @@ ADDRESS here is taken to mean the same thing as the public key hash."
 
   ;; inefficient clearing algorithm -- ok for now, improve later!
   (loop with mempool = cosi-simgen:*mempool*
-        with transactions = (cosi/proofs:block-transactions block)
+        with transactions = (mapcar #'hash:hash/256 (cosi/proofs:block-transactions block))
         with block-tx-count = (length transactions)
         ;; initially (format t "~%Transactions count in block: ~d" (length transactions))
         ;;           (cosi/proofs/newtx:dump-txs :block block)
         for tx being each hash-value
           of mempool 
             using (hash-key key)
-        when (member tx transactions :test #'eq)
+        when (member (hash:hash/256 tx) transactions :test #'hash:hash=)
```

Comment:
OK, so the logic of this change working is that the transactions in the block are not necessarily EQ to the transactions that are in the mempool.  I thought they were.  In get-transactions-for-new-block, the transactions are simply pulled out of the mempool to create the list of transactions, and that's what makes up the list of transactions in the block.  I need to figure out: how could they end up with a different identity?  Also, I understood in the code in question, it's only run in the case of the leader.

Reply:
@markhdavid In every node other than the leader, the block containing the transactions has been de-serialized from the network.  Therefore `eq`, which tests object identity for the references to the same block of memory, isn't going to be equal.